File: leetcode-04/solutions/python3/0049.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ans = []
        dict = collections.defaultdict(list)

        for str in strs:
            s = ''.join(sorted(str))
            dict[s].append(str)

        for value in dict.values():
            ans.append(value)

        return ans

File: leetcode-04/solutions/cpp/0049.cpp
class Solution {
 public:
  vector<vector<string>> groupAnagrams(vector<string>& strs) {
    vector<vector<string>> ans;
    unordered_map<string, vector<string>> map;

    for (string& str : strs) {
      string s = str;
      sort(s.begin(), s.end());
      map[s].push_back(str);
    }

    for (auto& [key, value] : map) {
      vector<string> s = value;
      sort(s.begin(), s.end());
      ans.push_back(s);
    }

    return ans;
  }
};